apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -666,7 +666,7 @@ where
                     ShellParams {
                         block_gas_meter: &mut gas_meter,
                         write_log: &mut write_log,
-                        storage: &self.storage,
+                        storage: &mut self.storage,
                         wasm_dir: self.wasm_dir.as_path(),
```

Comment:
TODO: `wasm_dir` is no longer necessary. `storage` should still be able to be passed as immutable during a dry_run_tx or query.

Reply:
Being fixed up in https://github.com/anoma/namada/pull/310